src/instructlab/sdg/generate_data.py

```diff
@@ -78,33 +78,77 @@ def _get_response(logger, synth_example):
     return parts[1].strip() if len(parts) == 2 else parts[0].strip()
 
 
-def _gen_train_data(logger, output_datasets, output_file_train):
+def _convert_to_messages(sample):
+    """
+    Convert a sample dictionary to contain 'messages' and 'metadata' columns required for training.
+    """
+    # Create user query message
+    user_query = sample["inputs"]
+    # TODO: in the future we can remove the combinecolumnsblock and combine them here for simplicity
+    # if "context" in sample:
+    #     user_query = f"{sample['context']}\n\n{sample['inputs']}"
+
+    sample["messages"] = [
+        {"content": user_query, "role": "user"},
+        {"content": sample["targets"], "role": "assistant"},
+    ]
+    metadata = {
+        key: value
+        for key, value in sample.items()
+        if key not in ["messages", "inputs", "targets"]
+    }
+    sample["metadata"] = json.dumps(metadata)
+
+    # keeping required keys for messages training format
+    sample = {"messages": sample["messages"], "metadata": sample["metadata"]}
+
+    return sample
+
+
+def _gen_train_data(
+    logger, machine_instruction_data, output_file_train, output_file_messages
+):
     train_data = []
-    for output_dataset in output_datasets:
+    messages_data = []
+
+    for output_dataset in machine_instruction_data:
         for synth_example in output_dataset:
             logger.debug(synth_example)
             user = _get_question(logger, synth_example)
             if len(synth_example.get("context", "")) > 0:
                 user += "\n" + synth_example["context"]
-            train_data.append(
-                {
-                    "system": _SYS_PROMPT,
-                    "user": _unescape(user),
-                    "assistant": _unescape(_get_response(logger, synth_example)),
-                }
-            )
+            assistant = _unescape(_get_response(logger, synth_example))
+            train_entry = {
+                "system": _SYS_PROMPT,
+                "user": _unescape(user),
+                "assistant": assistant,
+            }
+            train_data.append(train_entry)
+            sample = {
+                "inputs": _unescape(user),
+                "targets": assistant,
+                "system": _SYS_PROMPT,
+            }
+            messages_data.append(_convert_to_messages(sample))
 
     with open(output_file_train, "w", encoding="utf-8") as outfile:
         for entry in train_data:
             json.dump(entry, outfile, ensure_ascii=False)
             outfile.write("\n")
 
+    with open(output_file_messages, "w", encoding="utf-8") as outfile:
```

Comment:
There is a simpler way to do this, the dataset is an HF Dataset - you do something like: 

```
train_data.to_json(outfile, orient='records', lines=True)
```

Reply:
Right now its a list of dictionaries, but we can use HF datasets too

Reply:
Yes please, we are generating HF datasets from the sdg pipeline - doesn't make sense to convert them back to list of dicts. 

Also the `_get_train_data` and `_get_test_data` is unnecessary - there are cleaner ways to do it using `.train_test_split(test_size=<split_size>)`

Reply:
here train and test data split is not happening from the same dataset, the test data is the leaf node data that is being used for inference purposes

Reply:
We dont need test dataset in messages format as that split will happen after data mixing, removing the test dataset

Reply:
We can also handle this in a follow-up in https://github.com/instructlab/sdg/issues/100